pkg/install/v1alpha1/cstor_volume.go

```diff
@@ -621,7 +621,12 @@ spec:
     Calculate the replica count
     Add as many poolUid to resources as there is replica count
     */}}
-    {{- $poolUids := keys .ListItems.cvolPoolList.pools | randomize }}
+    {{- $replicaAffinityName:= .TaskResult.creategetpvc.replicaAntiAffinity -}}
+    {{- if ne $replicaAffinityName  "none" -}}
+    {{- $poolUids := keys .ListItems.cvolPoolList.pools -}}
```

Comment:
This cvolPoolList seems to be coming from poolsList in RunTask `cstor-volume-create-listcstorpoolcr-default` as
`
    {{- $poolsList := jsonpath .JsonResult "{range .items[*]}pkey=pools,{@.metadata.uid}={@.metadata.name};{end}" | trim | default "" | splitListTrim ";" -}}
`
If there are evicted or terminating or containerCreating pool pods, does the above might give wrong output?

Reply:
cc: @AmitKumarDas

Reply:
You are correct @vishnuitta 
I guess `cspFilter` should take care of these stuff.
We might also need to ensure that filtered pools >= desired replicaCount
cc @ashishranjan738

Reply:
The above can be bug actually

Reply:
we do this fix in seperate PR